### PR TITLE
Update CreatePersistentVolumeClaimController to use getPreferredVersion

### PR DESCRIPTION
**Pull request by benjaminapetersen**
No view updates.

@spadgett

### CODE REVIEW

File: app/scripts/controllers/createPersistentVolumeClaim.js [hunk 1]
```diff
@@ -24,6 +25,8 @@ angular.module('openshiftConsole')
     $scope.accessModes="ReadWriteOnce";
     $scope.claim = {};
 
+    var perpersistentVolumeClaimsVersion = APIService.getPreferredVersion('buildconfigs');
```
Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: s/buildconfigs/persistentvolumeclaims

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: thx @spadgett, updating

**Comment by benjaminapetersen** [comment]
updated.

File: app/scripts/controllers/createPersistentVolumeClaim.js [hunk 2]
```diff
@@ -24,6 +25,8 @@ angular.module('openshiftConsole')
     $scope.accessModes="ReadWriteOnce";
     $scope.claim = {};
 
+    var perpersistentVolumeClaimsVersion = APIService.getPreferredVersion('persistentvolumeclaims');
```
Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: perper? :)

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: oh good grief, not cool.  ok, my review process sucks. lemme fix my system, sorry ☹️

Comment (spadgett) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: To be fair, I missed that on first review :)

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: at least var name consistently wrong but not broken, heh....  Updating.

**Comment by benjaminapetersen** [comment]
ok updated w/fixed var name.

**Comment by spadgett** [comment]
[merge]

**Comment by benjaminapetersen** [comment]
Updated w/`objectToResourceGroupVersion`.

File: app/scripts/controllers/createPersistentVolumeClaim.js [hunk 3]
```diff
@@ -49,7 +52,7 @@ angular.module('openshiftConsole')
       .then(_.spread(function(project, context) {
         $scope.project = project;
 
-        if (!AuthorizationService.canI('persistentvolumeclaims', 'create', $routeParams.project)) {
+        if (!AuthorizationService.canI(persistentVolumeClaimsVersion, 'create', $routeParams.project)) {
```
Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: We should be checking the same resourceGroupVersion that we create

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: Yeah that makes sense. Let me update.

**Comment by benjaminapetersen** [comment]
updated

File: app/scripts/controllers/createPersistentVolumeClaim.js [hunk 4]
```diff
@@ -34,6 +35,21 @@ angular.module('openshiftConsole')
       }
     ];
 
+    var pvcTemplate = {
+      kind: "PersistentVolumeClaim",
+      apiVersion: "v1",
+      metadata: {
+        name: $scope.claim.name,
```
Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: `$scope.claim.name` is not set at this point.

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: Oop, thx, fixed & tested.

**Comment by benjaminapetersen** [comment]
updated

File: app/scripts/controllers/createPersistentVolumeClaim.js [hunk 5]
```diff
@@ -59,7 +75,8 @@ angular.module('openshiftConsole')
           if ($scope.createPersistentVolumeClaimForm.$valid) {
             $scope.disableInputs = true;
             var claim = generatePersistentVolumeClaim();
-            DataService.create('persistentvolumeclaims', null, claim, context)
+            var createClaimVersion = APIService.objectToResourceGroupVersion(claim);
```
Comment (spadgett) [inline app/scripts/controllers/createPersistentVolumeClaim.js]: I'd use the `createPVCVersion` from above since we already have it

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: Yeah doesn't really make sense to do the work again, should still have the same `rgv` value.  ok will update.

Comment (benjaminapetersen) [reply app/scripts/controllers/createPersistentVolumeClaim.js]: Going to move it up as well, most of these I have near the top of the file, before the `ProjectService.get()`

**Comment by benjaminapetersen** [comment]
Updated/tested.

**Comment by spadgett** [comment]
[merge]

**Comment by spadgett** [comment]
flake #1684

[merge]

**Pull request merged**